tests/listobjects/listobjects.go

```diff
@@ -104,25 +105,36 @@ func runTests(t *testing.T, schemaVersion string, client ListObjectsClientInterf
 					require.NoError(t, err)
 				}
 
-				for _, assertion := range stage.Assertions {
+				for _, assertion := range stage.ListAssertions {
+					var contextualTuples *pb.ContextualTupleKeys
+					if len(assertion.ContextualTuples) > 0 {
+						contextualTuples = &pb.ContextualTupleKeys{
+							TupleKeys: assertion.ContextualTuples,
+						}
+					}
 					resp, err := client.ListObjects(ctx, &pb.ListObjectsRequest{
 						StoreId:          storeID,
 						Type:             assertion.Request.Type,
 						Relation:         assertion.Request.Relation,
 						User:             assertion.Request.User,
-						ContextualTuples: assertion.Request.ContextualTuples,
+						ContextualTuples: contextualTuples,
 					})
 
 					if assertion.ErrorCode == 0 {
 						require.NoError(t, err)
-						require.Subset(t, assertion.Expectation, resp.Objects)
+						require.ElementsMatch(t, assertion.Expectation, resp.Objects,
```

Comment:
note that the original UT infrastructure has bugs in that it cannot detect expected element in list that is not in the actual list.

Reply:
This was there because if `ListObjectsMaxResults = 5` and `len(expected) = 6` and `len(actual) = 5`, we could not do a strict equality check

Reply:
But why would expected be 6 if LIstObjectsMaxResults is 5?  Should we create a unique test so that the expected = 5?

Reply:
Otherwise, there is a risk that the UT does not truly detect if list object misses some of the expected tuples

Reply:
> But why would expected be 6 if LIstObjectsMaxResults is 5?

Because there are 6 possible results that can be returned, and we cannot predict which of those will be returned since neither algorithm is deterministic

Reply:
I would argue that these are more exception rather than the rule ... we can use the subset specifically for these tests.  For normal test, we definitely should use exact match.